src/link/add.rs

```diff
@@ -753,6 +753,14 @@ impl LinkAddRequest {
             .append_nla(Nla::IfName(name))
     }
 
+    /// Create a wireguard link.
+    /// This is equivalent to `ip link add NAME type wireguard`.
+    pub fn wireguard(self, name: String) -> Self {
+        let mut request = self.name(name).link_info(InfoKind::Wireguard, None);
+        request.message_mut().header.flags &= !IFF_UP;
```

Comment:
The `LinkAddRequest::up` already exists. Please do not repeat there.

Reply:
...and this change is undoing it. Wireguard link is expected to be down.